parquet/src/test/java/org/apache/iceberg/parquet/TestPruneColumns.java

```diff
@@ -270,4 +273,64 @@ public void testStructElementName() {
     MessageType actual = ParquetSchemaUtil.pruneColumns(fileSchema, projection);
     assertThat(actual).as("Pruned schema should be matched").isEqualTo(expected);
   }
+
+  @Test
+  public void testVariant() {
+    MessageType fileSchema =
+        Types.buildMessage()
+            .addField(
+                Types.primitive(PrimitiveTypeName.INT32, Type.Repetition.REQUIRED)
+                    .id(1)
+                    .named("id"))
+            .addField(
+                Types.buildGroup(Type.Repetition.OPTIONAL)
+                    .as(LogicalTypeAnnotation.variantType((byte) 1))
+                    .addField(
+                        Types.primitive(PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED)
+                            .named("metadata"))
+                    .addField(
+                        Types.primitive(PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED)
+                            .named("value"))
+                    .id(2)
+                    .named("variant_1"))
+            .addField(
+                Types.buildGroup(Type.Repetition.OPTIONAL)
+                    .as(LogicalTypeAnnotation.variantType((byte) 1))
+                    .addField(
+                        Types.primitive(PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED)
+                            .named("metadata"))
+                    .addField(
+                        Types.primitive(PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED)
+                            .named("value"))
+                    .id(3)
+                    .named("variant_2"))
```

Comment:
Nit: This test is a bit hard to parse due to all the fields that need to be setup for a variant, could we abstract that behind a buildVariant helper

Reply:
I separated into helper method.